src/xpath.c

```diff
@@ -3980,6 +3980,83 @@ xpath_current(struct lyxp_set **args, uint32_t arg_count, struct lyxp_set *set,
     return LY_SUCCESS;
 }
 
+/**
+ * @brief Executes deref function on specific type and value. It performs evaluation in recursive manner,
+ *        which supports usage of union types as deref targets. Returns LYXP_SET_NODE_SET with either
+ *        leafref or instance-identifier target node(s).
+ *
+ * @param[in] leaf The target deref data node
+ * @param[in] sleaf The target deref schema node
+ * @param[in] value The currently evaluated value of target deref node depending on current type
+ * @param[in] cur_type The currently evaluated type of target deref node
+ * @param[in] top_type Whether the cur_type is the top level deref target node type or not
```

Comment:
Would call this `log` instead because that is the effect of this flag, an error is logged.